### PR TITLE
Cluster scoped quota

### PR DESCRIPTION
**Pull request by ncdc**
## Summary
Support quota of cluster-scoped things.

To use:
1. Create a namespace called `admin`
2. Create a ResourceQuota in `admin` with the following annotation: `"experimental.quota.kcp.dev/cluster-scoped": "true"`
3. You can now specify `count/<resource>` of both namespace- and cluster-scoped resources
    1. For namespace-scoped resources, it counts instances across all namespaces in the workspace
    2. For cluster-scoped resources, it counts instances in the workspace 

## Related issue(s)

Requires: https://github.com/kcp-dev/kubernetes/pull/97
Part of: #1061

### CODE REVIEW

File: pkg/reconciler/kubequota/scoped_generic_shared_informer_factory.go [hunk 1]
```diff
@@ -192,8 +192,19 @@ type scopedGenericNamespaceLister struct {
 // List lists all instances from the cache.Indexer scoped to a single logical cluster and namespace, and matching
 // selector.
 func (s *scopedGenericNamespaceLister) List(selector labels.Selector) (ret []runtime.Object, err error) {
```
Comment (stevekuznetsov) [inline pkg/reconciler/kubequota/scoped_generic_shared_informer_factory.go:194]: Who else uses this?

**Review by stevekuznetsov — COMMENTED** [review]
If I have a cluster-wide quota for configmaps of 5 and a namespace-specific quota of 3, will this correctly fail if I attempt to create a 4th configmap in one namespace? What determines quota precedence when multiple apply?

**Comment by ncdc** [comment]
> If I have a cluster-wide quota for configmaps of 5 and a namespace-specific quota of 3, will this correctly fail if I attempt to create a 4th configmap in one namespace? What determines quota precedence when multiple apply?

I'm pretty sure it fails if any matching ResourceQuotas fail

File: pkg/reconciler/kubequota/scoped_generic_shared_informer_factory.go (hunk 1 above)
Comment (ncdc) [reply pkg/reconciler/kubequota/scoped_generic_shared_informer_factory.go:194]: Nobody yet, but I could see us using it for GC

**Pull request merged**